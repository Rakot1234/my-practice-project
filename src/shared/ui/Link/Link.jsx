import React, { PureComponent } from 'react';
import './Link.scss';
import PropTypes from 'prop-types';

class Link extends PureComponent {
    static propTypes = {
        href: PropTypes.string.isRequired,
        className: PropTypes.string,
        hoverColor: PropTypes.oneOf(['yellow', 'sandy']),
        onClick: PropTypes.func,
        title: PropTypes.string.isRequired
    };

    static defaultProps = {
        className: '',
        hoverColor: 'yellow'
    };

    render() {
        const {
            href,
            className,
            hoverColor,
            onClick,
            title
        } = this.props;

        return (
            <span className={`link`}>
                <a
                    href={href} 
                    className={`link__element ${className} link__element-${hoverColor}`}
                    onClick={onClick}
                >
                    {title}
                </a>
            </span>
        ); 
    }
};

export default Link;
