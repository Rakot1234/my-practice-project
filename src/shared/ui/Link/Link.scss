@import '../../../styles/variables.scss';

.link {
    display: inline-block;
    &__element {
        //hover colors
        &-yellow:hover {
            color: #{yellow};
        }
        &-sandy:hover {
            color: #{$sandy};
        }
    }
};
