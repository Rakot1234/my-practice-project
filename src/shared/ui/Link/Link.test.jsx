import React from 'react';
import { shallow } from 'enzyme';
import Link from './Link';

const props = {
    title: 'Ссылка',
    href: '/'
};

describe('<Link />', () => {
    it('It render Link', () => {
        const wrapper = shallow(<Link {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Calls click handler', () => {
        const onClick = jest.fn();
        const wrapper = shallow(<Link {...props} onClick={onClick} />);
        wrapper.find('.link__element').simulate('click');
        expect(onClick).toHaveBeenCalled();
    })
});
