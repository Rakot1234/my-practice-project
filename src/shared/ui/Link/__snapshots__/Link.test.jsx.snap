// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> It render Link 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Link
    className=""
    hoverColor="yellow"
    href="/"
    title="Ссылка"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        className="link__element link__element-yellow"
        href="/"
      >
        Ссылка
      </a>,
      "className": "link",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Ссылка",
        "className": "link__element link__element-yellow",
        "href": "/",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "Ссылка",
      "type": "a",
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          className="link__element link__element-yellow"
          href="/"
        >
          Ссылка
        </a>,
        "className": "link",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Ссылка",
          "className": "link__element link__element-yellow",
          "href": "/",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Ссылка",
        "type": "a",
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
